core/src/main/scala/spire/random/Dist.scala

```diff
@@ -138,7 +143,7 @@ trait Dist[@sp A] extends Any { self =>
     rawHistogram(n).map { case (k, v) => (k, 1.0 * v / n) }
 
   def rawHistogram(n: Int)(implicit gen: Generator): Map[A, Int] =
-    toStream(gen).take(n).foldLeft(Map.empty[A, Int]) { case (h, a) =>
+    toLazyList(gen).take(n).foldLeft(Map.empty[A, Int]) { case (h, a) =>
```

Comment:
I'm not sure if there's value in constructing a `LazyList` over this or if `toIterator` would be more appropriate, since it's not being cached (unless I've missed something)

Reply:
No clue honestly, but using `toLazyList` seems safer.